core/src/commonMain/kotlin/com/ustadmobile/core/controller/ContentEntryEdit2Presenter.kt

```diff
@@ -318,23 +350,19 @@ class ContentEntryEdit2Presenter(context: Any,
 
         var entry: ContentEntryWithLanguage? = null
         try {
-            val metadata = contentImportManager?.extractMetadata(uri)
-            view.entryMetaData = metadata
-            when (metadata) {
-                null -> {
-                    view.showSnackBar(systemImpl.getString(MessageID.import_link_content_not_supported, context))
-                }
-            }
 
-            entry = metadata?.contentEntry
-            val entryUid = arguments[ARG_ENTITY_UID]
-            if (entry != null) {
-                if (entryUid != null) entry.contentEntryUid = entryUid.toString().toLong()
-                view.fileImportErrorVisible = false
-                if (metadata?.mimeType?.startsWith("video/") == true &&
-                        !metadata.uri.lowercase().startsWith("https://drive.google.com")) {
-                    view.videoUri = uri
-                }
+            val processContext = ProcessContext(createTemporaryDir("content"), mutableMapOf())
+            val metadata = pluginManager.extractMetadata(DoorUri.parse(uri), processContext)
+                    ?: throw IllegalArgumentException("no metadata found")
+            view.metadataResult = metadata
+            val plugin = pluginManager.getPluginById(metadata.pluginId)
+            entry = metadata.entry
+            arguments[ARG_ENTITY_UID]?.let { entry.contentEntryUid = it.toLong() }
```

Comment:
.let should be used to capture a result - e.g.

```
val something = someOtherVar?.let { it.lowercase() }
```

You could use .also or a normal if for this situation.